File: RemoveDuplicatesfromSortedArray.cpp
#include <bits/stdc++.h>
using namespace std;

int removeDuplicates(vector<int> &nums)
{
  if (nums.empty())
    return 0;
  int n = nums.size();
  int k = 0;
  for (int i = 0; i < n; i++)
  {
    if (nums[i] != nums[k])
    {
      k++;
      nums[k] = nums[i];
    }
  }
  return k + 1;
}

int main()
{
  vector<int> nums = {1, 1, 2};
  cout << removeDuplicates(nums) << endl;
  return 0;
}

File: remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {

  let i = 0;
  let j = 1;

  while (j < nums.length) {
    if (nums[i] !== nums[j]) {
      i++;
      nums[i] = nums[j];
    }
    j++;
  }

  console.log(nums);

  return i + 1;

};
let nums = [1, 1, 2];
console.log(removeDuplicates(nums));